src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -883,7 +883,7 @@ private static ValueExpression GetDeserializeImplementation(TypeProvider impleme
                 case Resource { ResourceData: SerializableObjectType resourceDataType } resource:
                     return New.Instance(resource.Type, new MemberExpression(null, "Client"), SerializableObjectTypeExpression.Deserialize(resourceDataType, element));
 
-                case MgmtObjectType mgmtObjectType when TypeReferenceTypeChooser.HasMatch(mgmtObjectType.ObjectSchema):
+                case MgmtObjectType mgmtObjectType when TypeReferenceTypeChooser.HasMatch(mgmtObjectType.InputModel):
```

Comment:
I should have seen exactly same code in `JsonCodeWriteExtensions`, could we add some todo to mark it as "to be refactored" in case we forget it?

Reply:
not sure which piece of code you are talking about, could you share a link to it?

Reply:
Here: https://github.com/Azure/autorest.csharp/blob/ced9b4b84e81a2daac4f7033ffd3693cd179ac67/src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs#L160

Reply:
Created https://github.com/Azure/autorest.csharp/issues/4777 for it.